File: frontend/server/contexts/user/domain/value-objects/password.ts
export interface Password {
  value: string;
}

export function definePassword(value: string): Password {
  if (!validate(value)) {
    throw new Error("Invalid password");
  }

  return {
    value,
  };
}

function validate(value: string) {
  const minLength = 6;

  return value.length >= minLength;
}

File: server/internal/user/domain/value_objects/password.go
package valueobjects

import "errors"

var (
	ErrInvalidPassword = errors.New("invalid password")
)

type Password string

func (ov Password) String() string {
	return string(ov)
}

func ValidatePassword(ov Password) error {
	return nil
	// return ErrInvalidPassword
}

func NewPassword(ov Password) (*Password, error) {
	if err := ValidatePassword(ov); err != nil {
		return nil, err
	}

	return &ov, nil
}
